A2017/tensorflow.ipynb

```diff
@@ -0,0 +1,892 @@
+    "        # Cutting through the error message...\n",
+    "        print('\\n'.join(str(e).split('\\n')[-3:]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Its value must be _explicitly_ passed to `session.run` via the `feed_dict` argument, which expects a `dict` mapping tensors to their value:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The value for p is 42.0\n",
+      "The value for p is 21.0\n"
+     ]
+    }
+   ],
+   "source": [
+    "with tf.Session() as session:\n",
+    "    # feed p the value 42\n",
+    "    print('The value for p is {}'.format(session.run(p, feed_dict={p: 42})))\n",
+    "    # feed p the value 21\n",
+    "    print('The value for p is {}'.format(session.run(p, feed_dict={p: 21})))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Random\n",
+    "\n",
+    "Another useful input tensor to have in our toolbox is the random tensor. The random seed can be set globally via `tf.set_random_seed`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "tf.set_random_seed(1234)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "There are many random distributions to choose from in TensorFlow. Let's look at `tf.random_uniform`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "r = tf.random_uniform(\n",
+    "    shape=[], minval=0.0, maxval=1.0, dtype=tf.float32, name='r')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A random tensor's value changes randomly between `session.run` calls, but the sequence of those random values stays the same across different sessions:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The value for r is 0.8478444814682007\n",
+      "The value for r is 0.23446130752563477\n",
+      "The value for r is 0.8478444814682007\n",
+      "The value for r is 0.23446130752563477\n"
+     ]
+    }
+   ],
+   "source": [
+    "with tf.Session() as session:\n",
+    "    print('The value for r is {}'.format(session.run(r)))\n",
+    "    print('The value for r is {}'.format(session.run(r)))\n",
+    "\n",
+    "with tf.Session() as session:\n",
+    "    print('The value for r is {}'.format(session.run(r)))\n",
+    "    print('The value for r is {}'.format(session.run(r)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Combining tensors\n",
```

Comment:
In this section, maybe should you mention that some ops have overloaded operators (e.g. `+` for tf.add()).
```python
a = tf.constant(1)
b = tf.constant(2)
c = a + b
```

Reply:
Done via 5a22e52.